checkov/terraform/graph_builder/variable_rendering/renderer.py

```diff
@@ -323,28 +323,41 @@ def _process_dynamic_blocks(dynamic_blocks: list[dict[str, Any]] | dict[str, Any
             if not block_content or not dynamic_values:
                 continue
 
-            dynamic_value_ref = f"{block_name}.value"
+            dynamic_value_dot_ref = f"{block_name}.value"
+            dynamic_value_bracket_ref = f'{block_name}["value"]'
             dynamic_arguments = [
                 argument
                 for argument, value in block_content.items()
-                if value == dynamic_value_ref or isinstance(value, str) and dynamic_value_ref in value
+                if (value == dynamic_value_dot_ref or value == dynamic_value_bracket_ref) or isinstance(value, str) and dynamic_value_dot_ref in value
```

Comment:
is it worth to create a tuple and check, if `value` is inside that tuple from a performance perspective 🤔 
```python
dynamic_value_refs = (dynamic_value_dot_ref, dynamic_value_bracket_ref)
```
```suggestion
                if (value in dynamic_value_refs or isinstance(value, str) and dynamic_value_dot_ref in value
```